Civi/Api4/Generic/Traits/DAOActionTrait.php

```diff
@@ -253,7 +253,7 @@ protected function formatCustomParams(&$params, $entityId) {
       }
 
       // Null and empty string are interchangeable as far as the custom bao understands
-      if (NULL === $value) {
+      if (NULL === $value || empty($value)) {
```

Comment:
@kurund I think this test failure relates:
``api\v4\Custom\FalseNotEqualsZeroTest::testFalseNotEqualsZero
Failed asserting that null is not null.``

I don't think it's right to use `empty()` here because what about the number 0? Even the string '0' is considered empty by php and we shouldn't change that to ''.

Reply:
@colemanw 

Agree, I have updated the code to be more specific to checkbox.